lib/types/svg.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+var fs      = require('fs');
+var Promise = require('bluebird');
+var pfstat  = Promise.promisify(fs.fstat);
+var pread   = Promise.promisify(fs.read);
+var utils   = require('../utils');
+
+var BUFFER_SIZE     = 8;
+var DETECTOR        = /^\s*</;
+var ROOT_MATCHER    = /<svg\s[^>]+>/;
+var WIDTH_MATCHER   = /\bwidth=(['"])([^%]+?)\1/;
+var HEIGHT_MATCHER  = /\bheight=(['"])([^%]+?)\1/;
+var VIEWBOX_MATCHER = /\bviewBox=(['"])(.+?)\1/;
+
+function isSVG (buffer) {
+  return utils.detectEncoding(buffer) === 'utf8' && DETECTOR.test(utils.ascii(buffer));
+}
+
+function parseAttributes (root) {
+  var widthMatches = root.match(WIDTH_MATCHER);
+  var heightMatches = root.match(HEIGHT_MATCHER);
+  var viewboxMatches = root.match(VIEWBOX_MATCHER);
+  return {
+    width: widthMatches && parseInt(widthMatches[2], 10),
+    height: heightMatches && parseInt(heightMatches[2], 10),
+    viewbox: viewboxMatches && parseViewbox(viewboxMatches[2])
+  };
+}
+
+function parseViewbox (viewbox) {
+  var bounds = viewbox.split(' ');
+  return {
+    width: parseInt(bounds[2], 10) - parseInt(bounds[0], 10),
+    height: parseInt(bounds[3], 10) - parseInt(bounds[1], 10)
+  };
+}
+
+function findRoot (fd, size, offset, string) {
+  if (offset >= size) {
+    return Promise.reject(new Error('Invalid SVG file'));
+  }
+
+  return pread(fd, new Buffer(BUFFER_SIZE), 0, BUFFER_SIZE, offset)
+  .spread(function (bytesRead, buffer) {
+    string += utils.ascii(buffer, 0, bytesRead);
+    var rootMatches = string.match(ROOT_MATCHER);
```

Comment:
RegExp match is done for the whole part read from the file on every `fs.read` pass.